### ISSUE TITLE
net8 support ?

### ISSUE THREAD

**Issue by hflexgrig** [jamesmontemagno/InAppBillingPlugin#623]
I'm using net8 for MAUI, and version 7.1.3 seems not working because of net7 dependency. Net8 release planned soon?

**Comment by hflexgrig** [comment]
Seems like we're waiting for this PR to be merged https://github.com/xamarin/GooglePlayServicesComponents/pull/815

**Comment by phillippschmedt** [comment]
We are using the plugin with .NET 8 on iOS and Android and it works just fine.

**Comment by hflexgrig** [comment]
> We are using the plugin in version 7.1.3 with .NET 8 on iOS and Android and it works just fine.

Once adding <PackageReference Include="Plugin.InAppBilling" Version="7.1.3" /> to my project, It not building, getting this at build errors

[{
	"resource": "/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.113/tools/Xamarin.Android.Common.targets",
	"owner": "msCompile",
	"code": "XARDF7024",
	"severity": 8,
	"message": "System.IO.IOException: Directory not empty : '/Users/haykgrigoryan/Projects/MathifyNew/Mathify.MathEditor/Mathify.MathEditor/obj/Debug/net8.0-android/lp/110/jl' [/Users/haykgrigoryan/Projects/MathifyNew/Mathify.MathEditor/Mathify.MathEditor/Mathify.MathEditor.csproj::TargetFramework=net8.0-android]",
	"source": "cpp",
	"startLineNumber": 2582,
	"startColumn": 2,
	"endLineNumber": 2582,
	"endColumn": 2
}]

I'm using mac M1 ARM architecture, not sure if it matters.

**Comment by phillippschmedt** [comment]
Hi @hflexgrig  I figured out I had to add some package downgrades to my csproj to make it work on our project, add the following to your csproj will hopefully fix the problem:

`    <ItemGroup Condition="'$(TargetFramework)' == 'net8.0-android'">
        <PackageReference Include="Xamarin.AndroidX.Activity" Version="1.9.0.3" /> 
        <PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.9.0.3" /> 
        <PackageReference Include="Xamarin.AndroidX.Collection" Version="1.4.0.5" /> 
        <PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.4.0.4" /> 
        <PackageReference Include="Xamarin.GooglePlayServices.Base" Version="118.4.0" />
    </ItemGroup>`

I also attached you an example project that builds just fine.
[InAppBillingTest.zip](https://github.com/user-attachments/files/16637003/InAppBillingTest.zip)

**Comment by hflexgrig** [comment]
@phillippschmedt Thanks a lot! now it builds, if I remove 
`<meta-data
		android:name="com.google.android.play.billingclient.version"
		android:value="7.0.0" />`

part from android manifest file. In your attached test project this part is missing in anroid manifest file. It is not needed anymore ?

**Comment by phillippschmedt** [comment]
I am not sure if it's needed. I never had that part in our manifest and it works for me.

**Comment by hflexgrig** [comment]
> Hi @hflexgrig I figured out I had to add some packages to my csproj to make it work on our project, add the following to your csproj will hopefully fix the problem:
> 
> ```
> <ItemGroup Condition="'$(TargetFramework)' == 'net8.0-android'">  
>         <PackageReference Include="Xamarin.AndroidX.Activity" Version="1.9.0.3" />   
>         <PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.9.0.3" />   
>         <PackageReference Include="Xamarin.AndroidX.Collection" Version="1.4.0.5" />   
>         <PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.4.0.4" />   
>         <PackageReference Include="Xamarin.GooglePlayServices.Base" Version="118.4.0" />  
> </ItemGroup>
> ```
> 

Btw after adding those into my csproj file, solution builds, but it getting stuck on attaching debugger infinitely.

**Comment by hflexgrig** [comment]
> I am not sure if it's needed. I never had that part in our manifest and it works for me.

This is reference from documentation https://jamesmontemagno.github.io/InAppBillingPlugin/GettingStarted.html

Android
For Xamarin.Android apps, in version 4 we use Xamarin.Essentials so you must ensure you initialize it in your Android project. It is setup by default in new projects:

protected override void OnCreate(Bundle savedInstanceState) {
    //...
    base.OnCreate(savedInstanceState);
    Xamarin.Essentials.Platform.Init(this, savedInstanceState); // add this line to your code, it may also be called: bundle
    //...
This is not needed in .NET 6 or .NET MAUI apps as we use built in .NET MAUI APIs to handle this for us.

You may also need to add the following metadata into your AndroidManifext.xml

<meta-data
            android:name="com.google.android.play.billingclient.version"
            android:value="6.1.0" />
Ensure the 6.1.0 matches the version of billing library you are using.

**Comment by phillippschmedt** [comment]
I don't believe the debugger issues are related to the mentioned packages since it works for me with those packages. I can't comment on your question regarding the billing version, apart from that we don't use it and billing works fine for me. That question needs to be answered by someone like @jamesmontemagno.

**Comment by hflexgrig** [comment]
> I don't believe the debugger issues are related to the mentioned packages since it works for me with those packages. I can't comment on your question regarding the billing version, apart from that we don't use it and billing works fine for me. That question needs to be answered by someone like @jamesmontemagno.

It started work! thanks a lot, you're a man!

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/4319050e-a701-4348-860a-26c0a4116adc">

**Comment by jamesmontemagno** [comment]
You don't need to add the billing items into the manifest as it does it automatically with the base packages. 

v7 is built against .NET 6 and Billing Library 6.X, which is why you may run into odd issues with the packages like that. 

This being said, i would checkout 8.0 beta which is a nice re-implementation with minor api chagnes and is built against .NET 8: https://www.nuget.org/packages/Plugin.InAppBilling/8.0.3-beta
